### DISCUSSION TITLE
Snapshot Collector dll is causing a "Very High" security flaw with Veracode application security platform

### DISCUSSION THREAD

**Discussion by chrisztjx** [microsoft/ApplicationInsights-SnapshotCollector#26] [Q&A]
Hi folks,

We have an application using the latest version of the Snapshot Collector package, and .NET 8.0, which we are running [Veracode](https://www.veracode.com/) application security scans on. Those scans are reporting a "Very High" error within this package, citing this CWE: https://cwe.mitre.org/data/definitions/78.html , CWE 78: Improper Neutralization of Special Elements used in an OS Command ('OS Command Injection'). Veracode says that the location of this error is: microsoft_applicationinsights_snapshotcollector_dll.Microsoft.ApplicationInsights.SnapshotCollector.UploaderProcess .

Can this be addressed/fixed? Please let me know. 
Thanks!

**Comment by pharring** [comment]
Security issues and bugs should be reported privately to the Microsoft Security Response Center (MSRC), either by emailing [secure@microsoft.com](mailto:secure@microsoft.com) or via the portal at https://msrc.microsoft.com/. You should receive a response within 24 hours. If for some reason you do not, please follow up via email to ensure we received your original message. Further information, including the MSRC PGP key, can be found in the [MSRC Report an Issue FAQ](https://www.microsoft.com/en-us/msrc/faqs-report-an-issue).

Please do not open issues for anything you think might have a security implication.
